docs/external-node.md

```diff
@@ -0,0 +1,305 @@
+the `ExternalNode`. `antrea-controller`
+creates an `antreaagentinfo` object for each new `ExternalNode`, and then `antrea-agent`
+updates it every minute with its latest status. `antreaagentinfo` is deleted by
+`antrea-controller` when the `ExternalNode` is removed.
+
+## Traffic model
+
+### OVS pair ports
+
+`antrea-agent` uses the interface IPs or name to find the network interface on the external Node, and then attaches it
+on the OVS bridge. The network interface is attached on OVS as uplink, and a new OVS internal Port is added to take over
+the uplink interface's IP/MAC and routing configurations. On Windows, the DNS configurations are also moved to the OVS
+internal port from uplink. Before attaching the uplink to OVS, the network interface is renamed with a suffix
+"~", and OVS internal port is configured with the original name of the uplink. As a result, IP/MAC/routing entries are
+seen on a network interface configuring with the same name on the external Node.
+
+The outbound traffic sent from the external Node enters OVS from the internal port, and finally output from the uplink,
+and the inbound traffic enters OVS from the uplink and output to the internal port. The IP packet is processed by the
+OpenFlow pipeline, and the non-IP packet is forwarded directly.
+
+Following is a diagram for the traffic model on an external Node.
+![Traffic On ExternalNode](assets/traffic_external_node.svg)
+
+### OpenFlow pipeline
+
+A new OpenFlow pipeline is implemented by `antrea-agent` dedicated for `ExternalNode`
+feature.
+
+![OVS pipeline](assets/ovs-pipeline-external-node.svg)
+
+#### Non-IP packet
+
+`NonIPTable` is a new OpenFlow table introduced only when on an external Node, which is dedicated for all non-IP
+packets. Non-IP packet is forwarded between the pair ports directly, e.g., a non-IP packet entering OVS from the uplink
+interface is output to the paired internal port, and a packet from the internal port is output to the uplink.
+
+#### IP packet
+
+One OpenFlow pipeline is set on an external Node to process the IP packet. Antrea NetworkPolicy is the major function in
+this new pipeline, and the OpenFlow tables used are similar to the container case. **No L3 routing is provided on an
+external Node**, and a simple L2 forwarding policy is implemented. OVS connection track is used to assist the
+NetworkPolicy function, in a result only the first packet is validated by the OpenFlow entries, and the sequencing
+packets in an accepted connection are allowed directly .
+
+**Egress/Ingress Tables**
+
+Table `XgressSecurityClassifierTable` is installed in both `stageEgressSecurity` and
+`stageIngressSecurity`, which is used to install the OpenFlow entries to ensure a specific IP traffic to bypass Antrea
+NetworkPolicy rules, e.g.,the SSH connection from a specific host to the external Node. These rules can be set with the
+item
+`policyBypassRules` in the agent configuration. Following is a configuration example,
```

Comment:
We should explain this in a separate "user" section, as it is really what users would care. We should not hide it in openflow design.

Reply:
Or considering a sub-section under "Apply Antrea NetworkPolicy to ExternalNode"

Reply:
Move it as a sub-section under "Apply Antrea NetworkPolicy to ExternalNode"